File: my-folder/0967-minimum-falling-path-sum/solution.cpp
class Solution {
public:
    int minFallingPathSum(vector<vector<int>>& m) {
        int n = m.size();
        int minpath = INT_MAX;
      
        for(int i=1; i<n; i++) {
          for(int j=0; j<n; j++) {
              if(j==0) {
                  m[i][j] += min(m[i-1][j], m[i-1][j+1]);
              }
              else if(j==n-1) {
                m[i][j] += min(m[i-1][j-1], m[i-1][j]);
              }
              else {
                m[i][j] += min(min(m[i-1][j-1], m[i-1][j]), min( m[i-1][j], m[i-1][j+1] )); 
              }
          }
        }
        
        for( int j=0 ; j<n ; j++ ){
            minpath = min( minpath, m[n-1][j] );
        }

        return minpath ;


    }
};

File: my-folder/0967-minimum-falling-path-sum/solution.py
class Solution:
    def minFallingPathSum(self, matrix: List[List[int]]) -> int:
        n = len(matrix)
        m = len(matrix[0])

       
        for i in range(1, n):
            for j in range(0, m):
                if j == 0:
                    matrix[i][j] += min(matrix[i-1][j], matrix[i-1][j+1])
                elif j == (m-1):
                    matrix[i][j] += min(matrix[i-1][j-1], matrix[i-1][j])
                else:
                    matrix[i][j] += min( matrix[i-1][j-1], min( matrix[i-1][j], matrix[i-1][j+1] ) )
        mini = matrix[n-1][0]
        for k in range(0, m):
            mini = min(mini, matrix[n-1][k])
        return mini


